config/routes.rb

```diff
@@ -3,6 +3,10 @@
     root 'articles#index'
 
     resources :articles
+    resources :users
```

Comment:
So in this case you are using `resources :users` which will add all of the routes for `users`. Below two of these routes are being created specifically. So now we have 7 + 2 routes. I think it might be better to do one of the following:

`resources :users, only: [:show, :update, :destroy]` (obviously choosing which ones we want)

or

`resources :users, except: [:new, :create]` (exclusion is a bit easier if you want everything except your custom routes)

Reply:
I mean for now you can even remove `resources :users` since the only methods in the controller are the ones for `sign_up`.

Reply:
Keep in mind you also have to ensure that your `users_path` exists and right now the `resources :users` is taking care of that. The `users_path` is used by the `form_with model` helper.

It's possible that I'm wrong about this and it is quite common to declare all the resource routes and then also point additional pretty routes to the same controller methods.

I think this might be an important consideration with the request testing. When we use `users_path` or `users_url` helpers, what is this providing? It's going to provide `/users` (post users_url) or `/users/new` (get new_user_url). So that tells me that the tests aren't specifying that there is a `/sign_up` (get sign_up_url) or `/sign_up` (post sign_up_url` route and making requests to that. Nothing is verifying that the `/sign_up` route points to the `users#new` controller, or the other `/sign_up` route points the the `users#crate` controller.

So tough call if I would remove routes that I don't want to be used. It does create challenges when using helpers that assume a certain prefix. Just make sure to specify what the app should work like in your tests, and ensure you are testing that.